programs/ziti-edge-tunnel/instance.c

```diff
@@ -34,14 +36,40 @@ tunnel_identity *find_tunnel_identity(const char* identifier) {
     }
 }
 
-tunnel_identity *get_tunnel_identity(char* identifier) {
+tunnel_identity *create_or_get_tunnel_identity(char* identifier, char* filename) {
     tunnel_identity *id = find_tunnel_identity(identifier);
 
     if (id != NULL) {
+        if (filename != NULL) {
+            id->Status = instance_status_ok;
+        }
         return id;
     } else {
         tunnel_identity *tnl_id = calloc(1, sizeof(struct tunnel_identity_s));
         tnl_id->Identifier = strdup(identifier);
+        if (filename != NULL) {
+            char* extension = strstr(filename, ".json");
+
+            size_t length;
+            if (extension != NULL) {
+                length = extension - filename;
+            } else {
+                length = strlen(filename);
+            }
+            tnl_id->FingerPrint = calloc(length + 1, sizeof(char));
+            char fingerprint[FILENAME_MAX];
+            memcpy(fingerprint, filename, length);
+            fingerprint[length] = '\0';
+            snprintf(tnl_id->FingerPrint, length+1, "%s", fingerprint);
+
+            if (tnl_id->Name == NULL) {
```

Comment:
Is it necessary to test `tnl_id->Name` for NULL here? I think it will always be NULL, since `tnl_id` was just calloc'ed on line 48.